File: sample/dracula.java
import java.util.List;

record Vampire(String location, int birthDate, int deathDate, List<String> weaknesses) {
  int age() {
    return deathDate - birthDate;
  }
}

final class DraculaSample {
  public static void main(String[] args) {
    var dracula = new Vampire("Transylvania", 1428, 1476, List.of("Sunlight", "Garlic"));
    var alucard = new Vampire("Wallachia", 1458, 1510, List.of("Holy Water"));

    System.out.printf("Dracula: %d, Alucard: %d%n", dracula.age(), alucard.age());
  }
}

File: sample/dracula.go
package main

import "fmt"

type Vampire struct {
	Location   string
	BirthDate  int
	DeathDate  int
	Weaknesses []string
}

func (v Vampire) Age() int {
	return v.DeathDate - v.BirthDate
}

func main() {
	dracula := Vampire{
		Location:   "Transylvania",
		BirthDate:  1428,
		DeathDate:  1476,
		Weaknesses: []string{"Sunlight", "Garlic"},
	}

	alucard := Vampire{
		Location:   "Wallachia",
		BirthDate:  1458,
		DeathDate:  1510,
		Weaknesses: []string{"Holy Water"},
	}

	fmt.Printf("Dracula: %d, Alucard: %d\n", dracula.Age(), alucard.Age())
}
